guides/octane-auth-torii-with-github.md

```diff
@@ -0,0 +1,668 @@
+[Back to Main README](../README.md)
+
+# Full GitHub Authorization in Ember Octane with `torii-provider`
+
+This repository provides a minimal [Ember
+Octane](https://emberjs.com/editions/octane/) application for authentication
+and authorization with GitHub. It uses
+[`ember-simple-auth`](http://github.com/simplabs/ember-simple-auth) and
+[`torii`](http://github.com/Vestorly/torii) to provide the authentication and authorization.
+
+This guide is an update and reworking of [the guide provided by
+`ember-simple-auth`](https://github.com/simplabs/ember-simple-auth/blob/master/guides/auth-torii-with-github.md),
+but it uses Octane syntax and eschews jQuery. It also uses a production-ready
+gatekeeper server instead of an `http-mock` server to provide the token
+exchange. 
+
+The [basic flow of authenticating with
+GitHub](https://developer.github.com/apps/building-oauth-apps/authorizing-oauth-apps/)
+deserves an outline that is personalized for Ember:
+
+1. When the user clicks the login button, the Ember application contacts
+   `https://github.com/login/oauth/authorize` with your app’s `client_id`.
+   GitHub presents a new window asking the user to authorize access to their
+   GitHub account from your app. The user grants this.
+2. GitHub redirects back to your app with an `authorizationCode`. This is
+   proof of authorization, but your app still doesn’t know _who_ the user is
+   (so it cannot connect to a `User` model, for example).
+3. Your app sends the `authorizationCode` to a gatekeeper server you maintain,
+   which in turn `POST`s the code to
+   `https://github.com/login/oauth/access_token`. In development, this can be
+   mocked. In production, it must be a separate server, about which more
+   below, as we will use a separate server throughout.
+4. GitHub responds to the `POST` with an `access_token`, which your gatekeeper
+   server forwards to your app.
+5. Your app uses the `access_token` in an `Authorization: Bearer
+   <access_token>` header request to, say, `https://api.github.com/user`,
+   which returns the GitHub user’s user info, at which point you can match the
+   username with your `User` model or whatever. 
+
+As with the original `ember-simple-auth` guide, the approach here mirrors any
+OAuth2 explicit grant flow used also by Facebook, Google, etc. 
+
+You can find a sequence diagram for the flow [here](http://github.com/simplabs/ember-simple-auth/guides/assets/esa-initial-flow.txt.png).
+
+## Set up and Configuration
+
+This section of the guide is for newcomers to OAuth, GitHub, and Ember. For
```

Comment:
Personally, I think it would be nice to not create an arbitrary separation of the audience between "newbies who are in need of this guide"  and "experts who can skip it" and label them as "newbies" or "experts" depending on the fact if they find the TL;DR that is provided below clear already or not

I think people who read the TL;DR might in fact already be experts at Ember, OAuth and Github and still have more questions about the details of the setup which are explained in the extended guide. I think the TL;DR itself is super useful in any case, so what do you think about a more neutral intro along the lines of

```suggestion
The following guide will teach you about the basics of OAuth, Github and Ember, following these five major steps to implement Github Authorisation in your Ember app:
```

What are your thoughts?